### DISCUSSION TITLE
Semantic highlighting recognising magic method decorated with lru_cache as function

### DISCUSSION THREAD

**Discussion by Bobronium** [microsoft/pylance-release#2920] [Enhancement]
## Environment data

-   Language Server version: `2022.5.1 (pyright e0e07d31)`
-   OS and version: 12.3.1 (21E258)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10, pyenv

## Code Snippet

```python
from functools import lru_cache


class Foo:
    @lru_cache(maxsize=1)
    def __str__(self):
        return "magic"

```
<details>
  <summary>Screenshots</summary>
  
<img width="688" alt="image" src="https://user-images.githubusercontent.com/36469655/168492502-faa15304-c65c-4cf9-8325-378c69f84543.png">
<img width="584" alt="image" src="https://user-images.githubusercontent.com/36469655/168492548-33f7aadc-59e4-4964-9bf5-3cc0b47c440f.png">
<img width="755" alt="image" src="https://user-images.githubusercontent.com/36469655/168492658-7817b7ae-cd56-428a-99de-a773e79582cc.png">

  
</details>

## Repro Steps
1. Enable semanticTokenColorCustomizations by putting this in `settings.json`:
  ```js
      "editor.semanticTokenColorCustomizations": { "enabled": true },
  ```
2. Open code snippet in editor
3. Use `>Inspect Editor Tokens and Scopes` to spot the bug

## Expected behavior
Magic methods decorated with lru_cache show this info:
```
semantic token type | magicFunction
modifiers | declaration
```

## Actual behavior
Magic methods decorated with lru_cache show this info:
```
semantic token type | function
modifiers | decorator
```

## Logs
Too large to fit here. Will post if necessary.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Reply by Bobronium** [reply]
Isn't it a bug though? What's here to discuss 🤔
